scheduler-server/src/main/java/gov/nasa/jpl/aerie/scheduler/server/http/ActivityTemplateJsonParser.java

```diff
@@ -0,0 +1,61 @@
+package gov.nasa.jpl.aerie.scheduler.server.http;
+
+import gov.nasa.jpl.aerie.json.JsonParseResult;
+import gov.nasa.jpl.aerie.json.JsonParser;
+import gov.nasa.jpl.aerie.json.SchemaCache;
+import gov.nasa.jpl.aerie.merlin.protocol.types.SerializedValue;
+import gov.nasa.jpl.aerie.scheduler.server.models.SchedulingDSL;
+import gov.nasa.jpl.aerie.scheduler.server.services.MissionModelService;
+
+import javax.json.Json;
+import javax.json.JsonObject;
+import javax.json.JsonValue;
+import java.util.HashMap;
+import java.util.Map;
+
+public class ActivityTemplateJsonParser implements JsonParser<SchedulingDSL.ActivityTemplate> {
+
+  private final Map<String, MissionModelService.ActivityType> activityTypesByName = new HashMap<>();
+
+  public ActivityTemplateJsonParser(MissionModelService.MissionModelTypes activityTypes){
+    activityTypes.activityTypes().forEach((actType)-> activityTypesByName.put(actType.name(), actType));
+  }
+
+  @Override
+  public JsonObject getSchema(final SchemaCache anchors) {
+    return Json
+        .createObjectBuilder()
+        .add("type", "object")
+        .build();
+  }
+
+  @Override
+  public JsonParseResult<SchedulingDSL.ActivityTemplate> parse(final JsonValue json) {
+    if (!(json instanceof JsonObject)) return JsonParseResult.failure("expected object");
+    final var asObject = json.asJsonObject();
+    if(!asObject.containsKey("activityType") || !asObject.containsKey("args")) return JsonParseResult.failure("expected fields activityType and args");
+    final var activityType = asObject.getString("activityType");
+    final var args = asObject.getJsonObject("args");
+    final var map = new HashMap<String, SerializedValue>(json.asJsonObject().size());
+    for (final var field : args.entrySet()) {
+      final var parsedValue = new StrictSerializedValueJsonParser(this.activityTypesByName.get(activityType).parameters().get(field.getKey())).parse(field.getValue());
+      if(parsedValue.isFailure()){
+        return JsonParseResult.failure("error while parsing argument with key "+ field.getKey());
+      }
```

Comment:
Let's use `prependBreadcrumb` here to avoid losing the information from the `parsedValue` failure message.

```suggestion
      if (parsedValue instanceof JsonParseResult.Failure<SerializedValue> failure){
        failure.prependBreadcrumb(Breadcrumb.ofString(field.getKey()));
        return failure.cast();
      }
```

Reply:
Done